server/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -226,59 +222,26 @@ private boolean shouldBeReadByCurrentNode(URI uri) {
     private void initCurrentReader(FileInput fileInput, URI uri) throws IOException {
         lineProcessor.startWithUri(uri);
         InputStream stream = fileInput.getStream(uri);
-        currentReader = createBufferedReader(stream);
+        currentInputStream = new BufferedInputStream(stream);
         currentLineNumber = 0;
-        lineProcessor.readFirstLine(currentUri, inputFormat, currentReader);
+        lineProcessor.readFirstLine(currentUri, inputFormat); // No header processing for now, but stream has to be passed later to do that
     }
 
-    private void closeCurrentReader() {
-        if (currentReader != null) {
+    private void closeCurrentStream() {
+        if (currentInputStream != null) {
             try {
-                currentReader.close();
+                currentInputStream.close();
             } catch (IOException e) {
                 LOGGER.error("Unable to close reader for " + currentUri, e);
             }
-            currentReader = null;
+            currentInputStream = null;
         }
     }
 
-    private String getLine(BufferedReader reader, long startFrom, int retry) throws IOException {
-        String line = null;
-        try {
-            while ((line = reader.readLine()) != null) {
-                currentLineNumber++;
-                if (currentLineNumber < startFrom) {
-                    continue;
-                }
-                if (line.length() == 0) {
-                    continue;
-                }
-                break;
-            }
-        } catch (SocketTimeoutException e) {
-            if (retry > MAX_SOCKET_TIMEOUT_RETRIES) {
```

Comment:
component which actually reads the data has to take care of retries - so need to push this logic further down the pipeline. for now only happy path is tested